app/services/encrypted_email.rb

```diff
@@ -38,9 +36,11 @@ def fingerprint
   private
 
   attr_accessor :email, :encrypted_email
+  attr_writer :user_access_key
 
   def decrypt
     encryptor = Pii::PasswordEncryptor.new
-    encryptor.decrypt(encrypted_email, self.class.user_access_key)
+    self.user_access_key = self.class.user_access_key
```

Comment:
how about `new_user_access_key`?

the important distinction imo is that the class method returns a new instance, while the instance method returns the same UAK that was used to decrypt the email.